asus/zephyrus/ga402x/shared.nix

```
@@ -43,15 +58,20 @@ in {
             evdev:name:*:dmi:bvn*:bvr*:bd*:svnASUS*:pn*:*
             KEYBOARD_KEY_ff31007c=f20
           '';
-          extraRules = ''
-            # Disable auto-suspend for the ASUS N-KEY Device, i.e. USB Keyboard
-            # Otherwise, it will tend to take 1-2 key-presses to wake-up after suspending
-            ACTION=="add", SUBSYSTEM=="usb", TEST=="power/autosuspend", ATTR{idVendor}=="0b05", ATTR{idProduct}=="19b6", ATTR{power/autosuspend}="-1"
-          '';
         };
       };
     }
 
+    (mkIf (! cfg.keyboard.autosuspend.enable) {
+      services.udev = {
+        extraRules = ''
```

the `mkIf` could have been placed here in the prior version, causing a smaller diff overall
```suggestion
        extraRules = mkIf (! cfg.keyboard.autosuspend.enable) ''
```

---

I'll try to rework it a little further to incorporate this.

---

> the `mkIf` could have been placed here in the prior version, causing a smaller diff overall

Does this work better, @pbsds ?